File: c/01find_3_numbers_for_a_sum_to_0.c
#include<stdio.h>

int main()
{
	int a=0, b, c;
	int nums[5];
	for(int i=0; i<5; i++)
	{
		scanf("%d", &nums[i];
	}

	for(a=0; a<5; a++)
	{
		for(b=a+1; b<5-a; b++)
		{
			for(c=b+1; c<4-a; c++)
			{
				if(nums[a] ==0 - nums[b] - nums[c])
				{
					printf("%d %d %d", nums[a], nums[b], nums[c];
				}	
			
			}
		}
	}
	return 0;
}

File: Python/01find_3_numbers_for_a_sum_to_0.py
res=[]
a=input()
item=a.split(" ")
nums=[eval(i) for i in item]
nums.sort()

for i in range(len(nums)):
    if nums[i]>0:
        break
    if(i>=1 and nums[i]==nums[i-1]):
        continue
    target=-nums[i]
    j=i+1
    k=len(nums)-1
    while j<k:
        if nums[j]+nums[k]==-nums[i]:
            res.append(nums[i])
            res.append(nums[j])
            res.append(nums[k])
            while j<k and nums[j]==nums[j+1]:
                j+=1
            while j<k and nums[k]==nums[k-1]:
                k-=1
            j+=1
            k-=1
        elif nums[j]+nums[k]<-nums[i]:
            j+=1
        else:
            k-=1

print([res[i] for i in range(len(res))])

